### PR TITLE
Add information about after clause timeout limit

### PR DESCRIPTION
**Pull request by michalmuskala**
Source: http://erlang.org/doc/reference_manual/expressions.html#id80928

### CODE REVIEW

File: lib/elixir/lib/kernel/special_forms.ex [hunk 1]
```diff
@@ -1908,6 +1908,10 @@ defmodule Kernel.SpecialForms do
 
     * `0` - if there is no matching message in the mailbox, the timeout
       will occur immediately
+      
+  The timeout value must be a positive integer smaller than 0xffffffff or
```
Comment (eksperimental) [inline lib/elixir/lib/kernel/special_forms.ex]: The timeout value must be a **non-negative integer**

Comment (eksperimental) [inline lib/elixir/lib/kernel/special_forms.ex]: We need backticks around the numbers `0xffffffff`

File: lib/elixir/lib/kernel/special_forms.ex [hunk 2]
```diff
@@ -1908,6 +1908,10 @@ defmodule Kernel.SpecialForms do
...
+  4294967295 - it should be possible to represent the timeout value as an
```
Comment (eksperimental) [inline lib/elixir/lib/kernel/special_forms.ex]: I would separate that huge number with underscores as `4_294_967_295`

File: lib/elixir/lib/kernel/special_forms.ex (hunk 1 above)
Comment (lexmag) [inline lib/elixir/lib/kernel/special_forms.ex]: Maybe the following will read better:
```
smaller than `4_294_967_295` (`0xFFFFFFFF` in hex notation)
```
_Note, I used uppercase in hex notation: https://github.com/lexmag/elixir-style-guide#hex-literals._

**Comment by josevalim** [comment]
Agreed with @lexmag.
-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

**Comment by michalmuskala** [comment]
Updated

**Comment by eksperimental** [comment]
sorry, but it still says that it must be a positive number, while `0` is still a valid timeout.

**Comment by michalmuskala** [comment]
You're right, I forgot about that change.

But actually thinking about it, we list "special values" just above - it includes zero. So I would say, zero is already handled in the definition. Maybe we should introduce it with "The regular timeout value" to differentiate from the "special values".

**Comment by eksperimental** [comment]
:+1 on differentiating special from regular values.

**Comment by michalmuskala** [comment]
I reworked this a bit more now.

The docs were incorrect since they said the timeout can be a variable, when in fact, any expression evaluating to the allowed values will do. I also changed the structure slightly so that we could merge the special value with the regular value definitions.

File: lib/elixir/lib/kernel/special_forms.ex [hunk 3]
```diff
@@ -1900,15 +1900,19 @@ defmodule Kernel.SpecialForms do
...
+  The timeout value given to `after` can be any expression evaluating to
+  one of the allowed values:
 
     * `:infinity` - the process should wait indefinitely for a matching
       message, this is the same as not using a timeout
 
     * `0` - if there is no matching message in the mailbox, the timeout
       will occur immediately
 
+    * positive integer smaller than `4_294_967_295` (`0xFFFFFFFF `
```
Comment (eksperimental) [inline lib/elixir/lib/kernel/special_forms.ex]: extra space after 0xFFFFFFFF

File: lib/elixir/lib/kernel/special_forms.ex [hunk 4]
```diff
@@ -1900,15 +1900,19 @@ defmodule Kernel.SpecialForms do
       end
 
   The `after` clause can be specified even if there are no match clauses.
-  The timeout value given to `after` can be a variable; two special
-  values are allowed:
+  The timeout value given to `after` can be any expression evaluating to
```
Comment (lexmag) [inline lib/elixir/lib/kernel/special_forms.ex]: So much better wording. 👏

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**